### PR TITLE
Rustup

### PR DESCRIPTION
**Pull request by Manishearth**
From https://github.com/rust-lang/rust/pull/66246/

changelog: none

### CODE REVIEW

**Comment by Manishearth** [comment]
@matthewjasper could you have a look at this? Specifically the "fix categorizations" commit, I'm not sure if that's the right fix for the cases that were recursively looping

Also, I'm not sure where I should be getting an infcx from, `tcx.infer_ctxt(|| { ... })`?

File: clippy_lints/src/escape.rs [hunk 1]
```diff
@@ -105,15 +103,15 @@ fn is_argument(map: &hir::map::Map<'_>, id: HirId) -> bool {
 }
 
 impl<'a, 'tcx> Delegate<'tcx> for EscapeDelegate<'a, 'tcx> {
-    fn consume(&mut self, cmt: &cmt_<'tcx>, mode: ConsumeMode) {
-        if let Categorization::Local(lid) = cmt.cat {
+    fn consume(&mut self, cmt: &Place<'tcx>, mode: ConsumeMode) {
+        if let PlaceBase::Local(lid) = cmt.base {
```
Comment (matthewjasper) [inline clippy_lints/src/escape.rs]: In this file there should also be a check that `cmt.projections` is empty.

Comment (flip1995) [reply clippy_lints/src/escape.rs]: I just added it to every method of the EUV. It didn't break anything, so I think this should be ok?

**Pull request merged**